src/app/projects/add-edit-project/add-edit-project.component.ts

```diff
@@ -549,19 +552,21 @@ export class AddEditProjectComponent implements OnInit, AfterViewInit, OnDestroy
    * @returns {FormGroup[]} The array of shapefile form groups.
    */
     buildShapefilesFormArray(projectData: Project): FormGroup[] {
+      let shapefilesToFillFormWith = this.shapefileDocuments as unknown as ProjectShapefileOrDocument[];
+      
       if (Array.isArray(projectData.shapefiles)) {
-        return projectData.shapefiles.map(shapefile => {
-          return new FormGroup({
-            'document': new FormControl(shapefile.document),
-            'documentFileName': new FormControl(shapefile.documentFileName),
-            'title': new FormControl(shapefile.title),
-            'colour': new FormControl(shapefile.colour),
-            'order': new FormControl(shapefile.order)
-          })
-        })
-      } else {
-        return [];
+        shapefilesToFillFormWith = projectData.shapefiles;
       }
+
+      return shapefilesToFillFormWith.map(shapefile => {
+        return new FormGroup({
+          'document': new FormControl(shapefile?.document || shapefile?.id),
+          'documentFileName': new FormControl(shapefile.documentFileName),
```

Comment:
Just want to verify that this is the only field that is mandatory/expected

Reply:
It is! This function only runs if there are shapefiles to use. Both shapefile source types will have `documentFileName`